docs/contributor/arch/023-unstructured-log-parsing.md

```diff
@@ -0,0 +1,179 @@
+# 23. Unstructured Log Parsing
+
+Date: 2025-07-30
+
+## Status
+
+Proposed
+
+## Context
+
+The current Fluent Bit-based LogPipeline custom resources offer minimal log parsing capabilities. However, with the ongoing migration toward OpenTelemetry (OTel)-based logging, we aim to maintain feature parity while improving the user experience and flexibility of configuration.
+
+## Fluent Bit Log Parsing Overview
+
+Fluent Bit supports various parsing mechanisms:
+* Regex-based Parsing: Using the LogParser CR, applied to logs of specifically annotated workloads.
+* Multiline Parsing: Through custom multiline filter.
+* Built-in Parsers: For common log formats like JSON, Nginx, Apache, etc.
+
+Limitations of Fluent Bit Approach
+* Requires manual annotation of each pod to enable parsing.
+* Limited support for mixed log formats within the same workload (needs further validation).
+* Uses inconsistent APIs for different parsing types, which can confuse users.
+
+## OTel Log Parsing: Requirements & Expectations
+
+The OTel-based solution should fulfill the following requirements:
+
+### Parser Definition
+Since LogPipeline have output (backend) affinity and parsers have input (workload) affinity, parsers must not be defined in the LogPipeline. Ideally, they should be defined in a separate resource. However, for now, we want to avoid introducing a new API and define parsers in the Telemetry CR.
```

Comment:
```suggestion
Because LogPipeline resources have output (backend) affinity and parsers have input (workload) affinity, parsers must not be defined in the LogPipeline. Ideally, they should be defined in a separate resource. However, for now, we want to avoid introducing a new API, and thus define parsers in the Telemetry CR.
```